pkg/register/register.go

```diff
@@ -108,8 +108,13 @@ func (r *client) Register(reg elementalv1.Registration, caCert []byte, state *St
 			return nil, fmt.Errorf("failed to send SMBIOS data: %w", err)
 		}
 
-		if protoVersion >= MsgSystemData {
+		if protoVersion >= MsgSystemDataNG {
```

Comment:
Well, would be better if we rework slightly this section and move out the if/then/else branches from the the main register function. Something like  reworking the sendSystemData(conn, protover) function, passing the protocol version and managing there what we send (bios & system-data/system-data-v2).